src/runmode-dpdk.c

```diff
@@ -0,0 +1,1295 @@
+/* Copyright (C) 2021 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \ingroup dpdk
+ *
+ * @{
+ */
+
+/**
+ * \file
+ *
+ * \author Lukas Sismis <lukas.sismis@gmail.com>
+ *
+ * DPDK runmode
+ *
+ */
+
+#include "suricata-common.h"
+#include "runmodes.h"
+#include "runmode-dpdk.h"
+#include "source-dpdk.h"
+#include "util-runmodes.h"
+#include "util-byte.h"
+#include "util-dpdk.h"
+
+#ifdef HAVE_DPDK
+
+#define ROUNDUP(x, y) ((((x) + ((y)-1)) / (y)) * (y))
+
+#define RSS_HKEY_LEN 40
+static uint8_t rss_hkey[] = { 0x6D, 0x5A, 0x6D, 0x5A, 0x6D, 0x5A, 0x6D, 0x5A, 0x6D, 0x5A, 0x6D,
```

Comment:
not sure I fully understand the comment here, but in general with af-packet and others one of the major issues that ppl fight with is that the rss hashing isn't symmetric by default. So fixing that for users is something I'd be in favor of. That said, I think there are NIC specific options too? IIRC the i40e dpdk driver can set a 'rss symmetric' mode, so that'd mean we can leave the key untouched I think.

Reply:
I'm fine with this for this first round, but I would like to make sure this does get addressed in a future update. Ths RSS key trick feels like a hack and if we have devices that are capable of doing things properly we should.